src/rust_helpers/number.rs

```diff
@@ -0,0 +1,40 @@
+use std::hash;
+
+#[derive(Clone, Copy, Debug)]
+pub struct Number(f64);
+
+impl Number {
+    fn new(value: f64) -> Self {
+        if value.is_nan() {
```

Comment:
Per [this answer](https://stackoverflow.com/a/39647997/732366), this is basically manually implementing `PartialEq`/`Eq`/hashability for `f64` which I guess is ok as long as you aren't allowing `NaN` values. If we end up having to use this type to represent the JS/Typescript `NaN` value I'm probably picturing converting this to an enum eg `Number::NaN`/`Number::Number(f64)`. There could be other issues as far as representing JS/Typescript number values as an `f64` (vs per above comment in the Typescript version they can just use the "same" JS number type) but not sure